setup.py

```diff
@@ -96,6 +94,7 @@
           'sqlalchemy',
       ],
       extras_require={
+          'all': ['ginga', 'imexam'],
```

Comment:
Hmm, what about `ds9`/`pyds9`? Ideally you would require *either* `ds9` or `ginga` for interactive use, but that's impossible to automate unless one of them is already installed. Also, it looks like `imexam` itself has a hard dependency on `ginga` (but not `pyds9`)... And also on `ipython` and `notebook`, in fact, which seems a bit much just to get `ds9` interaction :slightly_frowning_face:. The biggest issue was just the unconditional `imexam` import in order to use `geminidr`, but it would also be nice if `imexam` didn't pull all this stuff in, then it could just be a hard dependency (though it would still be one of only a few deps from Astroconda).

Reply:
The dependencies for imexam seems limited to numpy/scipy/astropy/matplotlib: https://github.com/spacetelescope/imexam/blob/d6f21d13d0e46a07de21512c0a25d5a97084a138/setup.cfg#L23-L51
I don't see ginga or ipython/notebook ?

Reply:
Hmm, that's just what `conda info` was telling me, but it looks like they've got rid of the `ipython` dependencies in the latest Python 3 builds, sorry (still has `ginga` though).

Reply:
https://github.com/astroconda/astroconda-contrib/blob/master/imexam/meta.yaml

Reply:
Sorry about the eventual S/N in that paragraph...

Reply:
ginga seems optional (though missing in their setup.py) but conda does not have the concept of optional dependencies...

Reply:
Right, but/so usually if a dependency is optional, I would not include it in the conda package requirements. I'm not sure why they have done so here in that case.

Reply:
Maybe just to ensure that it has at least one back end available.

Reply:
Yeah I don't know, it seems to come from an old version: https://github.com/astroconda/astroconda-contrib/pull/90